src/main/java/picard/analysis/directed/RnaSeqMetricsCollector.java

```diff
@@ -95,21 +95,34 @@ public RnaSeqMetricsCollector(final Set<MetricAccumulationLevel> accumulationLev
         return ignoredSequenceIndices;
     }
 
-    private class PerUnitRnaSeqMetricsCollector implements PerUnitMetricCollector<RnaSeqMetrics, Integer, SAMRecord> {
+    protected class PerUnitRnaSeqMetricsCollector implements PerUnitMetricCollector<RnaSeqMetrics, Integer, SAMRecord> {
+
+        protected final RnaSeqMetrics metrics;
```

Comment:
:+1: thanks for the explanation!

On Tue, Jul 12, 2016 at 12:45 PM, alecw <notifications@github.com> wrote:

> In src/main/java/picard/analysis/directed/RnaSeqMetricsCollector.java
> <https://github.com/broadinstitute/picard/pull/597#discussion_r70474984>:
>
> > @@ -95,21 +95,34 @@ public RnaSeqMetricsCollector(final Set<MetricAccumulationLevel> accumulationLev
> >          return ignoredSequenceIndices;
> >      }
> >
> > -    private class PerUnitRnaSeqMetricsCollector implements PerUnitMetricCollector<RnaSeqMetrics, Integer, SAMRecord> {
> > +    protected class PerUnitRnaSeqMetricsCollector implements PerUnitMetricCollector<RnaSeqMetrics, Integer, SAMRecord> {
> > +
> > +        protected final RnaSeqMetrics metrics;
>
> Hi @yfarjoun <https://github.com/yfarjoun> ,
>
> I'm not sure I understand the question. The metrics member remains final
> in this PR. All I did was make it protected, because otherwise it wasn't
> accessible in a subclass in a different package.
>
> It isn't feasible (at least not easily) to have an overridable
> getMetrics(), because the subclass isn't fully initialized until after the
> base class ctor has returned, and as you can see, the metrics member is
> accessed in the base class ctor. I tried an overridable getMetrics()
> initially, and settled on what I did after discovering that getMetrics()
> wouldn't work. Ultimately I think what I've done is cleaner, because there
> aren't 2 metrics objects for each PerUnitCollector to create confusion.
>
> -Alec
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/broadinstitute/picard/pull/597/files/fa0d6859340a1e54feb916a149bc03e3ff85fc03#r70474984>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ACnk0hwLLURMT3cC4qvvMGbX_tWNzv0Yks5qU8SjgaJpZM4JKbbz>
> .
>